botx/clients/clients/sync_client.py

```diff
@@ -108,19 +108,38 @@ def execute(self, request: HTTPRequest) -> HTTPResponse:
                 headers=request.headers,
                 params=request.query_params,
                 json=request.json_body,
+                data=request.data,
+                files=request.files,
             )
         except httpx.HTTPError as httpx_exc:
             raise BotXConnectError(
                 url=request.url,
                 method=request.method,
             ) from httpx_exc
 
+        headers = dict(response.headers)
+
+        should_process_as_error = (
```

Comment:
For what? Please, give me case when should we use this?

Reply:
For handling statuses other than 400 - 599